phpseclib/Crypt/Base.php

```diff
@@ -2001,11 +2057,48 @@ function _createInlineCryptFunction($cipher_code)
      * for which $mode the lambda function was created.
      *
      * @access private
-     * @return &Array
+     * @return Array &$functions
      */
     function &_getLambdaFunctions()
     {
         static $functions = array();
         return $functions;
     }
+
+    /**
+     * Generates a digest from $bytes
+     *
+     * @see _setupInlineCrypt()
+     * @access private
+     * @param $bytes
+     * @return String
+     */
+    function _trapdoor($bytes)
```

Comment:
What about `_digest_dynamic_function`? idk that I like `_digest_unrolled_function` because that implies that that's the only optimization. It's the only one I mentioned and recalled when I made the last post but it's not the only one. Quoting the header for the function that makes the dynamic functions:

```php
     *    The main reason why can speed up things [up to 50%] this way are:
     *
     *    - using variables more effective then regular.
     *      (ie no use of expensive arrays but integers $k_0, $k_1 ...
     *      or even, for example, the pure $key[] values hardcoded)
     *
     *    - avoiding 1000's of function calls of ie _encryptBlock()
     *      but inlining the crypt operations.
     *      in the mode of operation for() loop.
     *
     *    - full loop unroll the (sometimes key-dependent) rounds
     *      avoiding this way ++$i counters and runtime-if's etc...
```

The name of that method is `_createInlineCryptFunction`.